Nodejs/Product/Nodejs/TypingsAcquisition.cs

```diff
@@ -0,0 +1,143 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System.Threading.Tasks;
+using Microsoft.VisualStudioTools.Project;
+using Microsoft.NodejsTools.Npm;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+
+using SR = Microsoft.NodejsTools.Project.SR;
+using System;
+using System.Security;
+
+namespace Microsoft.NodejsTools {
+    internal class TypingsAcquisition {
+        private const string TsdExe = "tsd.cmd";
+        private const string TypingsDirectoryName = "typings";
+
+        private readonly string _pathToRootNpmDirectory;
+        private readonly string _pathToRootProjectDirectory;
+        private readonly Lazy<HashSet<string>> _acquiredTypingsPackageNames;
+
+        public TypingsAcquisition(
+            string pathToRootNpmDirectory,
+            string pathToRootProjectDirectory) {
+            _pathToRootNpmDirectory = pathToRootNpmDirectory;
+            _pathToRootProjectDirectory = pathToRootProjectDirectory;
+
+            _acquiredTypingsPackageNames = new Lazy<HashSet<string>>(() => {
+                return new HashSet<string>(CurrentTypingsPackages(_pathToRootProjectDirectory));
+            });
+        }
```

Comment:
Let's say you open a solution with 20 projects - we're going spin up 20 tsd instances as the project is loading, all of which are potentially longer-running and are going to spike the CPU at the same time. Additionally, you might get into a situation where we're we're printing out 20 error messages about tsd not being found (no need to focus on that one just yet, though). The ideal solution would be to have a single always-on service (so we don't have to wait for separate node processes to start) which every project can call into, and that's what we're aiming for in Dev15. For this, however, I think that "good enough" means that we're okay with spinning up several instances of tsd, but just make sure that happens in series rather than in parallel.